### PR TITLE
[machine learning] update pytorch user guide with pytorch 2.1 details

### PR DESCRIPTION
**Pull request by snadampal**
*Issue #, if available:*

*Description of changes:*
update pytorch user guide with pytorch 2.1 details

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: machinelearning/pytorch.md [hunk 1]
```diff
@@ -5,11 +5,11 @@
...
 
 # How to use PyTorch on Graviton CPUs
-There are multiple levels of software package abstractions available: AWS DLC (Deep Learning Container, gives the best performance, comes with additional optimizations on top of the official wheels, and all the packages installed), Python wheel (easiest option to get the release features but multiple wheels to be installed), and the Docker hub images (comes with downstream experimental features). Examples of using each method are below.
+There are multiple levels of software package abstractions available: AWS DLC (Deep Learning Container, comes with all the packages installed), Python wheel (easier option for integrating pytorch inference into the existing service), and the Docker hub images (comes with downstream experimental features). Examples of using each method are below.
```
Comment (AWSjswinney) [inline machinelearning/pytorch.md:8]:
```suggestion
There are multiple levels of software package abstractions available: AWS DLC (Deep Learning Container, comes with all the packages installed), Python wheel (easier option for integrating pytorch inference into an existing service), and the Docker hub images (comes with downstream experimental features). Examples of using each method are below.
```

Would this be better?

File: machinelearning/pytorch.md [hunk 2]
```diff
@@ -191,8 +189,8 @@ sudo apt install -y scons cmake
 cd $HOME
 git clone https://github.com/ARM-software/ComputeLibrary.git
 cd ComputeLibrary
-git checkout v22.11
-scons Werror=1 -j8 debug=0 neon=1 opencl=0 os=linux openmp=1 cppthreads=0 arch=armv8.2-a multi_isa=1 build=native
+git checkout v23.05.1
+scons Werror=1 -j8 debug=0 neon=1 opencl=0 os=linux openmp=1 cppthreads=0 arch=armv8a multi_isa=1 build=native
```
Comment (AWSjswinney) [inline machinelearning/pytorch.md:193]: I don't know the context here, but do we want to go backward to `armv8-a` instead of `armv8.2-a`?

**Review by AGSaidi — APPROVED** [review]

File: machinelearning/pytorch.md [hunk 3]
```diff
@@ -164,9 +162,9 @@ always [madvise] never
 export THP_MEM_ALLOC_ENABLE=1
 ```
 
-4. Starting PyTorch 1.13.0 release, mkldnn (OneDNN) backend is enabled for 'matmul' operator. While mkldnn along with ACL provides the best performance across several tensor shapes, the runtime setup overhead may not be acceptable for smaller tensor shapes. For light weight models with smaller tensor, check the performance by disabling the mkldnn backend and switching back to openBLAS gemm kernels. This has shown improvement for shapes like "12x12x64:12x64x12:12x12x12" and "12x16x16:12x16x64:12x16x64".
+4. Starting PyTorch 1.13.0 release, mkldnn (OneDNN) backend is enabled for 'matmul' operator. While mkldnn along with ACL provides the best performance across several tensor shapes, the runtime setup overhead may not be acceptable for smaller tensor shapes. For use cases with fewer input tokens, check the performance by disabling the mkldnn backend and switching to openBLAS gemm kernels. This has shown improvement for shapes like "12x12x64:12x64x12:12x12x12" and "12x16x16:12x16x64:12x16x64".
```
Comment (DDxPlague) [inline machinelearning/pytorch.md:165]: "Starting _with the release of Pytorch 1.13.0_" flows a bit better

File: machinelearning/pytorch.md (hunk 2 above)
Comment (snadampal) [reply machinelearning/pytorch.md:193]: This is the minimum version, "multi_isa=1" takes care of bringing newer ISA features at runtime.

**Review by AWSjswinney — APPROVED** [review]

**Pull request merged**